src/main/java/net/minecraftforge/common/extensions/IForgeLivingEntity.java

```diff
@@ -0,0 +1,69 @@
+package net.minecraftforge.common.extensions;
+
+import net.minecraft.world.entity.LivingEntity;
+import net.minecraft.world.level.material.FluidState;
+import net.minecraft.world.phys.Vec3;
+import net.minecraftforge.common.ForgeMod;
+import net.minecraftforge.fluids.FluidType;
+
+public interface IForgeLivingEntity extends IForgeEntity
+{
+    default LivingEntity self()
+    {
+        return (LivingEntity) this;
+    }
+
+    @Override
+    default boolean canSwimInFluidType(FluidType type)
+    {
+        if (self().isSensitiveToWater() && type == ForgeMod.WATER_TYPE.get()) return false;
+        else return IForgeEntity.super.canSwimInFluidType(type);
+    }
+
+    /**
+     * Performs what to do when an entity attempts to go up or "jump" in a fluid.
+     *
+     * @param type the type of the fluid
+     */
+    default void jumpInFluid(FluidType type)
+    {
+        self().setDeltaMovement(self().getDeltaMovement().add(0.0D, (double)0.04F * self().getAttribute(ForgeMod.SWIM_SPEED.get()).getValue(), 0.0D));
+    }
+
+    /**
+     * Performs what to do when an entity attempts to go down or "sink" in a fluid.
+     *
+     * @param type the type of the fluid
+     */
+    default void sinkInFluid(FluidType type)
+    {
+        self().setDeltaMovement(self().getDeltaMovement().add(0.0D, (double)-0.04F * self().getAttribute(ForgeMod.SWIM_SPEED.get()).getValue(), 0.0D));
```

Comment:
Is there a reason (other than the patches didn't do so before) not to have this just be `getAttributeValue`?